articles/building-a-simple-angular-app/index.md

```diff
@@ -0,0 +1,361 @@
+
+We will start with the data part first, and later on, build the UI and its working components.
+
+#### Getting your Data ready
+
+* In your `src/app` folder, create a new interface file under the new folder `model`.
+```typescript
+    // animeInterface.ts
+
+    export interface AnimeInterface {
+      id: number;
+      iconUrl: string;
+      name: string;
+      description: string;
+      type: string;     // whether a series/movie/OVA.
+      status: string;
+      comments: string[];
+    }
+```
+* The next step would be to populate the data. We will create a *typescript* file in the `src` folder.
+```typescript
+    //db-data.ts
+
+    import {AnimeInterface} from './app/model/animeInterface';
+    export const ANIME: AnimeInterface[] = [
+        {
+        id: 2,
+        iconUrl: '...enter icon URL',
+        name: 'Akira',
+        description: '...enter description here',
+        type: 'movie',
+        status: 'completed',
+        comments: []
+      },
+    ];
+```
+* The const `ANIME` is an array of type `AnimeInterface`, which holds data about each show in `JSON` format.
+With the data ready, the next step is to display all the anime cards.
+
+#### Building the UI
+
+We will be using `Bootstrap v4.0` to keep our UI simple and clean. Use the bootstrap templates in your `index.html` file.
+
+##### The Layout
+Create a component `anime-list` from your terminal with the following command:
+
+```commandline
+    ng g c animeList
+```
+  This will generate a new component called `anime-list` under the `src/` folder. 
+  The component gets imported into the `declaration` array in `app.module.ts`.
+
+  The `anime-list` component will be used to display the anime list from `db.data.ts` in a grid manner.
+  ```typescript
+    //anime-list.component.ts
+
+    import { Component } from '@angular/core';
+    import {ANIME} from '../../db-data';
+
+    @Component({
+      selector: 'app-anime-list',
+      templateUrl: './anime-list.component.html',
+    })
+
+    export class AnimeComponent{
+      animes = ANIME;
+    }
+  ```
+
+  ```html
+  <!-- anime-list.component.html -->
+
+    <div class="container-fluid">
+      <div class="col">
+        <div class="row animeCard">
+          <app-anime-card class="col-sm-3 col-md-3 col-lg-3"
+                          *ngFor="let anime of animes;index as animeId"
+                          [anime]='anime'
+                          [animeId] = animeId>
+          </app-anime-card>
+        </div>
+      </div>
+    </div>
+  ```
+In the above *.ts* code, we are sending 
+data from `db.data.ts` to the template, from where we are 
+calling the *component* `anime-card`.
+
+Now, create a `anime-card` component, which will display the anime card 
+and provide a static router link to the *anime profile* section.
+```typescript
+  // anime-card.component.ts
+
+    export class AnimeComponent{
+      @Input()
+      anime: AnimeInterface;
+
+      @Input()
+      animeId: number;
+    }
```

Comment:
The component name 'AnimeComponent' throws an error, kindly modify it to `AnimeCardComponent`(line 165)

Reply:
It has been modified and I had missed a file, added that file too.